pkg/controller/workload/job/job_controller.go

```diff
@@ -477,9 +477,12 @@ func podsCount(jobSpec *batchv1.JobSpec) int32 {
 func generatePodsReadyCondition(job *batchv1.Job, wl *kueue.Workload) metav1.Condition {
 	conditionStatus := metav1.ConditionFalse
 	message := "Not all pods are ready or succeeded"
-	if podsReady(job) && wl.Spec.Admission != nil {
+	// Once PodsReady=True it stays as long as the workload remains admitted to
+	// avoid unnecessary flickering the the condition when the pods transition
+	// Ready to Completed.
+	if (podsReady(job) || apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadPodsReady)) && wl.Spec.Admission != nil {
```

Comment:
This should be slightly faster when a workload is evicted.
```suggestion
	if (podsReady(job) || wl.Spec.Admission != nil && apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadPodsReady)) {
```

Reply:
Didn't work in the suggested form, but updated as `wl.Spec.Admission != nil && (podsReady(job) || apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadPodsReady))` to check for admission field first.